tf2onnx/onnx_opset/tensor.py

```diff
@@ -1464,18 +1465,41 @@ def version_10(cls, ctx, node, **kwargs):
         # onnx output is [num_selected_boxes, 3], the meaning of last dim is [batch_index, class_index, box_index]
         # while tf's output is [num_selected_boxes]
         ctx.insert_new_node_on_input(node, "Unsqueeze", node.input[0], axes=[0])
-        ctx.insert_new_node_on_input(node, "Unsqueeze", node.input[1], axes=[0, 1])
+        input_score = ctx.insert_new_node_on_input(node, "Unsqueeze", node.input[1], axes=[0, 1])
         ctx.insert_new_node_on_input(node, "Cast", node.input[2], to=onnx_pb.TensorProto.INT64)
         # replace original node with nonmaxsurppress + slice + squeeze + cast
-        dtypes = [ctx.get_dtype(node.output[0])]
-        shapes = [ctx.get_shape(node.output[0])]
+        dtypes = [[ctx.get_dtype(output)] for output in node.output]
+        shapes = [[ctx.get_shape(output)] for output in node.output]
+        max_output_size = node.input[2]
+        utils.make_sure(len(node.inputs) <= 5 or int(node.inputs[5].get_tensor_value(False)) == 0,
+                        "soft_nms_sigma must be 0")
         ctx.remove_node(node.name)
-        new_nonmaxsurppress = ctx.make_node(node.type, node.input).output[0]
+        new_nonmaxsurppress = ctx.make_node(node.type, node.input[: 5]).output[0]
         slice_op = GraphBuilder(ctx).make_slice({"data": new_nonmaxsurppress,
                                                  "axes": [1], "ends": [3], "starts": [2]})
         squeeze_op = ctx.make_node("Squeeze", [slice_op], attr={"axes": [1]})
-        ctx.make_node("Cast", inputs=squeeze_op.output, attr={"to": onnx_pb.TensorProto.INT32},
-                      name=node.name, outputs=node.output, dtypes=dtypes, shapes=shapes)
+        if len(node.input) > 5:  # V5
+            ctx.make_node("Cast", inputs=squeeze_op.output, attr={"to": onnx_pb.TensorProto.INT32},
+                          outputs=[node.output[0]], dtypes=dtypes[0], shapes=shapes[0])
+            ctx.make_node("Gather", inputs=[input_score.input[0], squeeze_op.output[0]],
+                          outputs=[node.output[1]], dtypes=dtypes[1], shapes=shapes[1])
+        elif "pad_to_max_output_size" in node.attr:  # V4
```

Comment:
What happens if op=NMS_V5 and "pad_to_max_output_size"==True? We will not do this step?

Reply:
v5 is triggered via non_max_suppression_with_scores, within which the pad_to_max_output_size always set to False.